File: python/7 kyu/234.py
# https://www.codewars.com/kata/5d9f95424a336600278a9632/train/python

# Passed

from math import floor, log2

def powers(n):
    power_array = []
    while n != 0:
        nearest_exponent = floor(log2(n))
        value = 2 ** nearest_exponent
        power_array.append(value)
        n -= value
    
    power_array = power_array[::-1]
    return power_array

output = powers(99)
print(output)

File: nodejs/7 kyu/234.js
// https://www.codewars.com/kata/55466644b5d240d1d70000ba/train/javascript

// Passed

function candies(kids) {
  if (kids.length === 1 || kids.length === 0) {
    return -1;
  }

  let maxCandies = Math.max(...kids);
  let candiesToHandOut = kids.map((candyCount) => maxCandies - candyCount);
  let totalCandiesToHandOut = candiesToHandOut.reduce(
    (prev, curr) => prev + curr,
    0,
  );
  return totalCandiesToHandOut;
}

const output = candies([5, 8, 6, 4]);
console.log(output);